### ISSUE TITLE
Training parameters?

### ISSUE THREAD

**Issue by tmaidment** [aabzaliev/gestures-language#1]
Hello!

I am trying to reproduce the results from your paper, since I want to use this model for another task. After fixing some issues to get this repository working on my machine, I tried a few times to train using the `train.py` script. However, I am consistently unable to achieve any sort of reasonable performance on the validation set (max val accuracy is ~0.04). I have downloaded your embeddings, and confirmed that they in fact perform well for the various tasks located in "reproduce_tables", but the embeddings that I have produced locally do not perform well.

What were the training parameters used to generate the final set of embeddings? I presume that they're different than the ones contained in the current code base.

Thank you!
- Tristan

**Comment by aabzaliev** [comment]
Hi Tristan,

thanks for reaching out. By accuracy 0.04 you mean CLIP matching accuracy, i.e. how good a pose is matched with an utterance? This accuracy would depend on the batch size. I trained the model with very low batch size=2, so the random accuracy would be 50%, and I was able to achieve 64%. Do you maybe use large batch size, since you say the accuracy is 0.04? I had worse results with larger batch sizes, presumably because it is hard to match poses and texts when there are many of them. 
Also, the codebase is the same I used for training, no other parameters were used.

**Comment by tmaidment** [comment]
Thanks for pointing that out, I see you also mentioned it in the paper. The default batch size in train.py is set to 32, which is what I tried the first time. I have set it to 2, it's still training but the preliminary results look okay.
